classes/Access.php

```diff
@@ -356,6 +356,17 @@ public function updateLgcAccess($idProfile, $idTab, $lgcAuth, $enabled, $addFrom
             $whereClauses[] = ' `slug` LIKE "' . $slugLike . '"';
         }
 
+        // We do not use Tab::recursiveTab which is risky during an upgrade
+        $idParent = (int) Db::getInstance()->getValue('
+            SELECT `id_parent`
+            FROM `' . _DB_PREFIX_ . 'tab`
+            WHERE `id_tab` = "' . (int) $idTab . '"
+        ');
+        // Add the same rights to the parent tab (don't do it in case of rights deletion)
+        if ($enabled && $idParent) {
```

Comment:
You call only if `$enabled = true` ?
This function can't be used to disable rights as well during the upgrade process?

Reply:
Indeed, it seems that `$enabled`  is true when we want to ADD rights. But when we want to disable one, this seems a bad idea to also disable the rights of the parents tabs. You could looe access to tabs depending on the same parent.